CMakeLists.txt

```diff
@@ -147,7 +147,6 @@ if (ENABLE_ALL_THE_DEBUG_MACROS)
     include(all_the_debug_macros)
 endif(ENABLE_ALL_THE_DEBUG_MACROS)
 
-set(CMAKE_STAGING_PREFIX "${CMAKE_BINARY_DIR}/Root")
 set(CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/Root")
```

Comment:
This commit is incorrect. If we should be changing anything, it's that our install prefix really is `/`. It shouldn't really matter, but according to your commit message this path somehow ends up in the Ports ecosystem.

What you are most likely experiencing is that CMake combines the combined prefix (which is correct) with `DESTDIR`, which then ends up creating that double path. The solution to that is usually to override `install` to not pass `DESTDIR`.

Reply:
Why are you saying that this is incorrect?
CMake's documentation literally says that this is useful for cases that we don't have. There is no need for `CMAKE_STAGING_PREFIX` and we should only rely on `CMAKE_SYSROOT`.

Reply:
> CMake's documentation literally says that this is useful for cases that we don't have.

We don't have this case because we use _other_ flags in a wrong way. `CMAKE_INSTALL_PREFIX` should, to my knowledge, not reference the sysroot (8f060bed17736bfab947d0d83272e24b691f3577).

`CMAKE_SYSROOT` may look like a viable alternative at first glance, but it doesn't affect `install`, at least according to its own documentation.

I'll do a `build_all` run when I have a chance, I'm quite certain that a bunch of ports will start installing themselves into the hosts `/usr/local` once `CMAKE_STAGING_PREFIX` is removed from `CMakeToolchain.txt`.

Reply:
> We don't have this case because we use other flags in a wrong way.

I was referring to the fact that the documentation recommend using `CMAKE_STAGING_PREFIX ` when you have a read only `CMAKE_SYSROOT`, which we don't have regardless of other flags.

> `CMAKE_INSTALL_PREFIX` should, to my knowledge, not reference the sysroot

I agree with that and my patch doesn't modify this

> I'll do a build_all run when I have a chance, I'm quite certain that a bunch of ports will start installing themselves into the hosts /usr/local once CMAKE_STAGING_PREFIX is removed from CMakeToolchain.txt.

Now that you say that, I copied the first implementation of the libjxl's `package.sh` from another CMake based port and it provided a custom invocation for `make install` and it did break when I removed `CMAKE_STAGING_PREFIX`, but if you let CMake do the installation, everything go as expected.

Reply:
> I was referring to the fact that the documentation recommend using CMAKE_STAGING_PREFIX when you have a read only CMAKE_SYSROOT, which we don't have regardless of other flags.

Then the documentation is wrong. `CMAKE_SYSROOT` doesn't have any effect on where files are `install`ed, only `CMAKE_INSTALL_PREFIX` does. Our `CMAKE_INSTALL_PREFIX` is read-only, so we need a `CMAKE_STAGING_PREFIX` that points into our build directory instead.

Case in point: I applied your CMake patch and we instantly went from 19 ports failing to build to 106 ports failing to build (both of these numbers include ports that themselves don't install _and_ ports that now have dependencies that are not installable). Every port that I checked that fails now whereas it didn't before fails because it tries to install to `/usr/local/...` instead of `/path/to/serenity/Build/<arch>/Root/usr/local` (which obviously doesn't work, since one would have to have root privileges to install files there.

>> `CMAKE_INSTALL_PREFIX` should, to my knowledge, not reference the sysroot
> I agree with that and my patch doesn't modify this

Yes. I'm arguing that if anything should be changed here, it should be `CMAKE_INSTALL_PREFIX` or nothing. However, that is just a technicality, and doesn't seem to be needed right now, so I'd just drop that idea entirely for the time being.

Reply:
I'm still convinced that there is something fishy with these two (`CMAKE_STAGING_PREFIX` and `CMAKE_SYSROOT`), from my findings `CMAKE_SYSROOT` does impact the installation directory. But anyway, I don't have time for this, so I won't touch it now.

Hmm for the failings ports, we have this overload of install in most of our CMake ports. It causes issues with my patch, but the solution was simply to remove it (at least for highway and libjxl).
```bash
install() {
    run make "${installopts[@]}" install
}
```